nifi-nar-bundles/nifi-azure-bundle/nifi-azure-parameter-providers/src/main/java/org/apache/nifi/parameter/azure/AzureKeyVaultSecretsParameterProvider.java

```diff
@@ -0,0 +1,201 @@
+            .build();
+
+    public static final String PARAMETER_GROUP_NAME = "group-name";
+
+    private static final List<PropertyDescriptor> properties;
+
+    static {
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(AZURE_CREDENTIALS_SERVICE);
+        props.add(KEY_VAULT_URL);
+        props.add(GROUP_NAME_PATTERN);
+        properties = Collections.unmodifiableList(props);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public List<ParameterGroup> fetchParameters(final ConfigurationContext context) throws IOException {
+        final SecretClient secretClient = configureSecretClient(context);
+        final List<KeyVaultSecret> secrets = getAllSecrets(secretClient);
+        final List<ParameterGroup> groups = getParameterGroupsFromSecrets(context, secrets);
+        return groups;
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(final ConfigurationContext context, final ComponentLog verificationLogger) {
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+
+        try {
+            final List<ParameterGroup> parameterGroups = fetchParameters(context);
+            int parameterCount = 0;
+            for (final ParameterGroup group : parameterGroups) {
+                parameterCount += group.getParameters().size();
+            }
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.SUCCESSFUL)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation(String.format("Fetched secret keys [%d] as parameters, across groups [%d]",
+                            parameterCount, parameterGroups.size()))
+                    .build());
+        } catch (final Exception e) {
+            verificationLogger.error("Failed to fetch parameters", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(ConfigVerificationResult.Outcome.FAILED)
+                    .verificationStepName("Fetch Parameters")
+                    .explanation("Failed to fetch parameters: " + e.getMessage())
+                    .build());
+        }
+        return results;
+    }
+
+    private List<KeyVaultSecret> getAllSecrets(final SecretClient secretClient) {
+        final List<KeyVaultSecret> secrets = new ArrayList<>();
+
+        for (final SecretProperties secretProperties : secretClient.listPropertiesOfSecrets()) {
+            KeyVaultSecret secretWithValue = secretClient.getSecret(secretProperties.getName(), secretProperties.getVersion());
+            final String name = secretWithValue.getName();
+            final String value = secretWithValue.getValue();
+            secrets.add(secretWithValue);
+            getLogger().debug("Retrieved Secret [{}] with value [{}]", name, value);
```

Comment:
This log should be removed, Secret Values should never be logged.

Reply:
Ah my bad, I forgot to remove that after testing, will remove, thanks!